### ISSUE TITLE
Incompatible architecture on macOS Monterey (12.1)

### ISSUE THREAD

**Issue by jackcogdill** [python-greenlet/greenlet#284]
- greenlet   1.1.2
- Python 3.9.9
- macOS-12.1-arm64-64bit

## Description

I am trying to install `pynvim`, which uses `greenlet` as a dependency. I was getting error messages from `greenlet`, so I tried installing it manually with `pip`.

Based on the wheel filename `greenlet-1.1.2-cp39-cp39-macosx_12_0_arm64.whl`, it appears to have downloaded the correct architecture (arm64; not x86_64). However, the main error message (below) says `have 'x86_64', need 'arm64e'`:

`ImportError: dlopen(/opt/homebrew/lib/python3.9/site-packages/greenlet/_greenlet.cpython-39-darwin.so, 0x0002): tried: '/opt/homebrew/lib/python3.9/site-packages/greenlet/_greenlet.cpython-39-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e'))`

Please let me know if I am missing something, or how to debug further.

## Debug output

>$ pip3 install greenlet
DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
Collecting greenlet
  Downloading greenlet-1.1.2.tar.gz (91 kB)
     |████████████████████████████████| 91 kB 3.6 MB/s
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: greenlet
  Building wheel for greenlet (setup.py) ... done
  Created wheel for greenlet: filename=greenlet-1.1.2-cp39-cp39-macosx_12_0_arm64.whl size=92430 sha256=6112d7ee91f148d23a801d5962748288e6cf97464b186f3a26121b58e5698444
  Stored in directory: /Users/kiyoshi/Library/Caches/pip/wheels/eb/d5/66/f93f7a70f83c1f5513e377d401e0fea8bb6655b16f091339f4
Successfully built greenlet
Installing collected packages: greenlet
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
Successfully installed greenlet-1.1.2

>$ python3
Python 3.9.9 (main, Nov 21 2021, 03:16:13)
[Clang 13.0.0 (clang-1300.0.29.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import greenlet
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/homebrew/lib/python3.9/site-packages/greenlet/__init__.py", line 29, in <module>
    from ._greenlet import _C_API # pylint:disable=no-name-in-module
ImportError: dlopen(/opt/homebrew/lib/python3.9/site-packages/greenlet/_greenlet.cpython-39-darwin.so, 0x0002): tried: '/opt/homebrew/lib/python3.9/site-packages/greenlet/_greenlet.cpython-39-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/usr/local/lib/_greenlet.cpython-39-darwin.so' (no such file), '/usr/lib/_greenlet.cpython-39-darwin.so' (no such file)
>>>
>>>
>>> import platform
>>> platform.platform()
'macOS-12.1-arm64-64bit'
>>>

**Comment by jamadden** [comment]
Your output indicates that you're building greenlet from source, so other than making sure that you have the latest versions of `pip` and `setuptools`, installed there's not much I can suggest besides making sure that your compiler toolchain (Xcode et all) is capable of producing the correct type of executables (e.g., make sure that no environment variable such as `ARCHFLAGS` exists and specifies the wrong type of architecture; for example, `ARCHFLAGS=-arch x86_64` might result in this error).

**Comment by jackcogdill** [comment]
Ah, thank you for providing that insight; I should have read the output more carefully.

I was able to resolve the issue. The problem was that my Alacritty was built for x86, causing everything within it to be built on rosetta. I followed these [instructions](https://github.com/alacritty/alacritty/blob/master/INSTALL.md#universal-binary) to build a universal Alacritty binary, and then uninstalled greenlet from pip, purged the cache, and reinstalled it.
